File: week_1/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int get_change(void);

int main(void)
{
    int coins = 0;
    int change = get_change();
    // Check how many quarters
    while (change >= 25)
    {
        coins += 1;
        change -= 25;
    }
    // Check how many dimes
    while (change >= 10)
    {
        coins += 1;
        change -= 10;
    }
    // Check how many nickels
    while (change >= 5)
    {
        coins += 1;
        change -= 5;
    }
    // Check how many cents
    while (change >= 1)
    {
        coins += 1;
        change -= 1;
    }
    printf("%i\n", coins);
}

// Prompt user to write how much change he's owed
int get_change(void)
{
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);
    int cents = round(change * 100);
    return cents;
}

File: week_6/pset6/cash/cash.py
from cs50 import get_float

# Ask user for change
while True:
    change = get_float("Change owed: ")
    if change < 0:
        change = get_float("Change owed: ")
    elif change >= 0:
        break

coins = 0

# Round the change
change = round(int(change * 100))

# Calculate coins
while change > 0:
    while change >= 25:
        coins += 1
        change -= 25
    while change >= 10:
        coins += 1
        change -= 10
    while change >= 5:
        coins += 1
        change -= 5
    while change >= 1:
        coins += 1
        change -= 1

# Print answer
print(coins)
